register_maps/byte-pair-to-page-reg.py

```diff
@@ -0,0 +1,216 @@
+    """
+    
+    bit_locations: List[RegisterLocationInSubBlock]
+    default_value: str
+
+    def __post_init__(self):
+        if len(self.bit_locations) == 0:
+            raise ValueError('A parameter cannot exist without any bit locations.')
+
+    @classmethod
+    def from_swamp_parameter(cls, parameter_node):
+        # the defval_mask is relative to the register it is in
+        # so we construct the default value after deducing the n_bits and min_bits
+        locations =  [
+            RegisterLocationInSubBlock.from_swamp_register(regnode)
+            for i_chunk, regnode in parameter_node.items()
+        ]
+        defval = 0
+        curr_min_bit = 0
+        for regnode, location in zip(parameter_node.values(), locations):
+            chunk = ((regnode['defval_mask'] >> location.min_bit) & ((1 << location.n_bits) - 1))
+            defval |= (chunk << curr_min_bit)
+            curr_min_bit += location.n_bits
+        # store the default value as a string in binary, padded out to the full length
+        # again, hopefully making it easier for comparison to the manual
+        bit_rep = '0b'+f'{defval:0b}'.zfill(curr_min_bit)
+        return cls(bit_locations = locations, default_value = bit_rep)
+
+
+    def to_cpp(self):
+        """Convert this parameter into its C++ representation
+
+        The `Parameter` and `RegisterLocation` structs are defined in pflib/src/pflib/Compile.cxx
+        before any of the implementation headers are included.
+        """
+        return 'Parameter({'+', '.join(l.to_cpp() for l in self.bit_locations)+'}, '+self.default_value+')'
+
+
+@dataclass
+class SubBlock:
+    """A subblock has an address and a type specifying its parameters"""
+    address: int
+    type: str
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument('input', help='input yaml file with R0/R1 addresses')
+parser.add_argument('output', type=Path, help='output file with page/register addresses')
+parser.add_argument('--no-intermediate-yaml', action='store_true', help='do not dump intermediate page/register addresses into a yaml for comparison. Used in the CMakeLists.txt to avoid extraneous files.')
+args = parser.parse_args()
+
+with open(args.input) as f:
+    byte_pair_map = yaml.safe_load(f)
+
+# the SWAMP block names sometimes correspond with the
+# names of the subblock types in the manual (e.g. "Top", "DigitalHalf", "ReferenceVoltage")
+# but sometimes don't (e.g. "ch" instead of "ChannelWise")
+# I want to translate them so the page names inside of pflib can stay the same
+# names (which are closer to those listed in the manual tables
+blockname_to_subblock_type = {
+    'ch': 'CHANNELWISE',
+    'cm': 'CHANNELWISE',
+    'calib': 'CHANNELWISE',
+    'HalfWise': 'CHANNELWISE',
+    'Top' : 'TOP',
+    'DigitalHalf': 'DIGITALHALF',
+    'GlobalAnalog': 'GLOBALANALOG',
+    'MasterTdc': 'MASTERTDC',
+    'ReferenceVoltage': 'REFERENCEVOLTAGE'
+}
```

Comment:
Many of the page and parameter names have changed (e.g. `channel_N` is now `ch_N`).

This is the location where we could inject a renaming of the page/parameter names. I am not sure if that is a good idea though since it would be another area to maintain; however, it would allow us to align the names with whats in the manual which I would appreciate for consistency.